src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1115,13 +1101,53 @@ internal DependencyObject GetContainerForIndex(int index)
 		{
 			var item = ItemFromIndex(index);
 
-			var container = IsItemItsOwnContainerOverride(item)
-				? item as DependencyObject
-				: GetContainerForItemOverride();
+			DependencyObject elementTemplateRoot = DataTemplateHelper.ResolveTemplate(ItemTemplate, ItemTemplateSelector, item, null)?.LoadContentCached() as DependencyObject;
+
+			var isItemItsOwnContainer = IsItemItsOwnContainerOverride(item);
+			var isContainerFromTemplateRoot = !isItemItsOwnContainer
+				&& this is Primitives.Selector
+				&& IsItemItsOwnContainerOverride(elementTemplateRoot);
+
+			DependencyObject container;
+
+			if (isItemItsOwnContainer)
+			{
+				container = item as DependencyObject;
+
+				// The item here is left untouched, it's not being applied
+				// the ItemTemplate/ItemTemplateSelector
+			}
+			else if (isContainerFromTemplateRoot)
+			{
+				container = elementTemplateRoot;
+				SetContainerContent(container, item, isGeneratedContainer: true);
+			}
+			else
+			{
+				container = GetContainerForItemOverride();
+				SetContainerContent(container, elementTemplateRoot);
```

Comment:
You still seem to be setting `elementTemplateRoot` as the content here. I would've thought you'd want to ignore `elementTemplateRoot` on this code path.

Reply:
That's the issue I was looking for :) There are a bunch of screenshot differences for which there are no associatd unit tests, I'll add a few for that case.

Thanks!